File: My algorithms collection/Bellman-Ford Algorithm/bellman_ford.py
class Edge:
    def __init__(self, u, v, w):
        self.u = u
        self.v = v
        self.w = w

def bellman_ford(edges, V, src):
    distance = [float('inf')] * V
    distance[src] = 0

    for _ in range(V - 1):
        for edge in edges:
            if distance[edge.u] + edge.w < distance[edge.v]:
                distance[edge.v] = distance[edge.u] + edge.w

    for edge in edges:
        if distance[edge.u] + edge.w < distance[edge.v]:
            print("Graph contains negative weight cycle")
            return

    print("Vertex Distance from Source")
    for i in range(V):
        print(f"{i}\t{distance[i]}")


edges = [Edge(0, 1, -1), Edge(0, 2, 4), Edge(1, 2, 3), Edge(1, 3, 2), Edge(1, 4, 2), Edge(3, 2, 5), Edge(3, 1, 1), Edge(4, 3, -3)]
V = 5
src = 0
bellman_ford(edges, V, src)

File: My algorithms collection/Bellman-Ford Algorithm/bellman_ford.java
import java.util.*;

class Edge {
    int u, v, w;
    Edge(int u, int v, int w) {
        this.u = u;
        this.v = v;
        this.w = w;
    }
}

public class BellmanFord {
    void bellmanFord(List<Edge> edges, int V, int src) {
        int[] distance = new int[V];
        Arrays.fill(distance, Integer.MAX_VALUE);
        distance[src] = 0;

        for (int i = 1; i < V; ++i) {
            for (Edge edge : edges) {
                if (distance[edge.u] != Integer.MAX_VALUE && distance[edge.u] + edge.w < distance[edge.v]) {
                    distance[edge.v] = distance[edge.u] + edge.w;
                }
            }
        }

        for (Edge edge : edges) {
            if (distance[edge.u] != Integer.MAX_VALUE && distance[edge.u] + edge.w < distance[edge.v]) {
                System.out.println("Graph contains negative weight cycle");
                return;
            }
        }

        System.out.println("Vertex Distance from Source");
        for (int i = 0; i < V; ++i) {
            System.out.println(i + "\t" + distance[i]);
        }
    }

    public static void main(String[] args) {
        List<Edge> edges = Arrays.asList(
            new Edge(0, 1, -1), new Edge(0, 2, 4),
            new Edge(1, 2, 3), new Edge(1, 3, 2),
            new Edge(1, 4, 2), new Edge(3, 2, 5),
            new Edge(3, 1, 1), new Edge(4, 3, -3)
        );
        int V = 5;
        int src = 0;
        new BellmanFord().bellmanFord(edges, V, src);
    }
}
